app/uk/gov/hmrc/thirdpartyapplication/services/commands/grantlength/ChangeGrantLengthCommandHandler.scala

```diff
@@ -41,7 +41,10 @@ class ChangeGrantLengthCommandHandler @Inject() (
 
   private def validate(app: StoredApplication, cmd: ChangeGrantLength): Validated[Failures, Unit] = {
     Apply[Validated[Failures, *]].map(
-      cond((cmd.grantLength.period.getDays != app.grantLength), CommandFailures.GenericFailure(s"Grant length is already ${app.grantLength} days"))
+      cond(
+        (cmd.grantLength.period != app.refreshTokensAvailableFor),
+        CommandFailures.GenericFailure(s"Grant length is already ${app.refreshTokensAvailableFor}")
```

Comment:
I _think_ this will log "Grant length is already P547D" which may not be the most readable format

Reply:
Addressed